### ISSUE TITLE
Order of bookmarks gets intermixed

### ISSUE THREAD

**Issue by asdfkaba** [marcelklehr/floccus#113]
floccus 2.25, bookmarks app 0.11.0

No real issue, just wanted to ask if its normal behaviour that the bookmark order in my bookmark bar gets intermixed after the bookmarks get pushed to the server. It seems it is caused due to the local updates afterwards(which i saw in console).

Great addon but with screwing up my order in bookmark bar its sadly no real alternative to official google syncing for me. Hopefully i'm missing something.

best regards.

**Comment by marcelklehr** [comment]
This is currently unavoidable with the nextcloud bookmarks app, as it doesn't store information about the order of bookmarks, *but* work is being done on a pure WebDAV backend for floccus which would store your bookmarks in a custom file in your nextcloud and would thus be able to retain the order of your bookmarks. The relevant issue is https://github.com/marcelklehr/floccus/issues/86 and work is being done over at #112 :)

**Comment by asdfkaba** [comment]
Nice! Thanks for the information :)

**Comment by vansteeny** [comment]
After testing this issue I came to the conclusion that it isn't fixed for the WebDav option.  The correct order of the bookmarks is retained in the xml file when syncing to the server but 2 problems still remains.

1) When moving or re-ordering a bookmark(s) this isn't registered as a change when syncing to the server. Only when you add/delete a bookmark does the moved bookmark get updated to the server. It does then register the change/reordering correctly and saves the correct order in the xml file.
2) When syncing from the server to the client on an other pc the new bookmarks still get added alphabetically to that browser. I tested this on firefox and chrome and both have the same issue.

Excellent add-on for the rest.

My best regards

**Comment by marcelklehr** [comment]
@vansteeny Oh, this is an amazing observation! Thanks for taking the time to investigate this! The fix was [rather easy](https://github.com/marcelklehr/floccus/commit/991a694b198f6d8f19c1508e429737c5f57ed2d9), but tough to spot :)

**Comment by bernd-wechner** [comment]
As an aside be great if order were preserved when using the Nextcloud Bookmarks app. Currently I have higgledy piggledy ordering of all my bookmarks on numerous browsers I use.

**Comment by marcelklehr** [comment]
@bernd-wechner I suspect you're stilil using the legacy sync method, which doesn't preserve ordering. The new sync method does. [Here's](https://github.com/marcelklehr/floccus#switching-from-tag-based-to-folder-based-syncing) how you can switch ;)

**Comment by bernd-wechner** [comment]
@marcelkehr, thanks heaps. Laborious process but great to see folders now in. 

They arrive in the same order which is nice. 

I did try this with no luck mind you:

1) Move a bookmark on one browser (up in the ordered list)
2) Sync that browser
3) Sync another browser

and the bookmark did not move on the other browser. I checked the Bookmarks app and can see maybe why, the thing I moved was a subfolder. It was in the middle of the list and I like to put them at top. Which I did. But on the Bookmarks app, subfolders don't appear in the same list as bookmarks at all and so may not be subject to order?

**Comment by marcelklehr** [comment]
> don't appear in the same list as bookmarks at all and so may not be subject to order?

The order is not currently displayed in the web app, but it is possible to put folders and bookmarks in one order via the API. One shouldn't make mistakes when programming, though :see_no_evil:

**Comment by marcelklehr** [comment]
v3.2.7 has the fix for this, now. Feedback is appreciated :)

**Comment by vansteeny** [comment]
Hi. I re-tested the 3.2.7 release.

The order in the xml file is still correct and changes gets synced properly to server.  Also performing a reorder syncs properly to the server.

When syncing to the server everything goes as it should. But when you then sync a second time all the new bookmarks if any get added and the order gets retained. But if you have a folder that contains folders combined with bookmarks the bookmarks gets sorted before the folders.

For example : [BOOKMARK LAYOUT]
Folder ("Shops") - Bookmark ("Google") - Folder ("Downloads") - Bookmark ("Adobe")

1 sync everything gets propagated to the server correctly

After second sync operation it sorts everything like this :
Bookmark ("Google") - Bookmark ("Adobe") - Folder ("Shops") - Folder ("Downloads")

Bookmarks first then folders.  If you never combine content (bookmarks & folders) in the root or subfolders this issue will not bother you. 

Hope this helps makes sense of the issue. ;)

**Comment by marcelklehr** [comment]
@vansteeny Awesome! This was very helpful. Thanks :)

**Comment by marcelklehr** [comment]
v3.2.8 has the fix for this, now. :)

**Comment by Dreeg** [comment]
> v3.2.8 has the fix for this, now. :)

Hi @marcelklehr, thanks a lot in advance for your work!

I found this thread because as soon as I installed and configured I found myself in the same situation.
NC Bookmark app version: 12.0.0
Floccus browser extension version: V4.18.1

Until I (by accident) found the "manual order" setting that fixed (it seems to me). It's just not selected by default.

Meanwhile my bookmarks have all moved and messed up with "folders first, then bookmarks".
With a little patience I will put them in order.

I was wondering though: why is "manual order" not the active option by default?
The problem I think is on the NC Bookmark app but it seems to me are you also in the development? Or did I misunderstand?
